app/src/main/java/com/example/easycloset/Item.java

```diff
@@ -0,0 +1,89 @@
+package com.example.easycloset;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.parse.ParseClassName;
+import com.parse.ParseFile;
+import com.parse.ParseObject;
+import com.parse.ParseUser;
+
+import java.util.Date;
+
+@ParseClassName("Item")
+public class Item extends ParseObject {
+
+    private static final String KEY_COLOUR = "colour";
+    private static final String KEY_IMAGE = "image";
+    private static final String KEY_USER = "user";
+    private static final String KEY_CATEGORY = "category";
+
+    public String getColour() {
+        return getString(KEY_COLOUR);
+    }
+
+    public void setColour(String colour) {
+        put(KEY_COLOUR, colour);
+    }
+
+    public ParseFile getImage() {
+        return getParseFile(KEY_IMAGE);
+    }
+
+    public void setImage(ParseFile image) {
+        put(KEY_IMAGE, image);
+    }
+
+    public ParseUser getUser() {
+        return getParseUser(KEY_USER);
+    }
+
+    public String getCategory() {
+        return getString(KEY_CATEGORY);
+    }
+
+    public void setCategory(String category) {
+        put(KEY_CATEGORY, category);
+    }
+
+    public void setUser(ParseUser user) {
+        put(KEY_USER, user);
+    }
+
+    @NonNull
+    public static String calculateTimeAgo(Date createdAt) {
+
+        int SECOND_MILLIS = 1000;
+        int MINUTE_MILLIS = 60 * SECOND_MILLIS;
+        int HOUR_MILLIS = 60 * MINUTE_MILLIS;
+        int DAY_MILLIS = 24 * HOUR_MILLIS;
+
+        try {
+            createdAt.getTime();
+            long time = createdAt.getTime();
+            long now = System.currentTimeMillis();
+
+            final long diff = now - time;
+            if (diff < MINUTE_MILLIS) {
+                return "just now";
+            } else if (diff < 2 * MINUTE_MILLIS) {
+                return "a minute ago";
+            } else if (diff < 50 * MINUTE_MILLIS) {
+                return diff / MINUTE_MILLIS + " m";
+            } else if (diff < 90 * MINUTE_MILLIS) {
+                return "an hour ago";
+            } else if (diff < 24 * HOUR_MILLIS) {
+                return diff / HOUR_MILLIS + " h";
+            } else if (diff < 48 * HOUR_MILLIS) {
+                return "yesterday";
+            } else {
+                return diff / DAY_MILLIS + " d";
+            }
+        } catch (Exception e) {
```

Comment:
that's in general is a bad practice to catch all, because it really easy to miss something that breaks your app. What kind of exception are you expecting? Is it possible to prevent that exception from happening instead of catching it? E.g. if you afraid to divide by 0 then you can check for 0

Reply:
fixed